### DISCUSSION TITLE
Highlight line numbers

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#4005] [Ideas]
Suggesting to add option to highlight line numbers in column at left side of screen.
This would be a similar to the setting to highlight current line.
Would allow the typical options of foreground (line number text), background, bold, italic
Would be a separate setting because you can already change the color of the line number font and line number background column.

Thank You

**Comment by elextr** [comment]
There is nothing that modifys styles of the line number, so the capability provided by Scintilla is not used.

**Comment by advice2020** [comment]
@elextr 
Thank you for your response.
I apologize, not sure if I understand correctly, would you be able to clarify?

**Comment by elextr** [comment]
Saying the same thing in another order, IIUC Scintilla provides capabilities for styling line numbers, but nothing Geany does uses that styling.

Something that uses styling of line numbers needs to be defined and PR code (and documentation) needs to be written by "somebody" to add to Geany.

**Comment by eht16** [comment]
I had a look at the Scintilla documentation and I don't think it is directly possible to style the margins (remember, there are more margins than just the line numbers and if so, we probably need to style every margin of the current line to have it consistent) according to the current line.

What *might* be possible would be to adjust the style of the margins of the current line every time the current line is changed. This will cause some more CPU cycles required on every click or cursor movement.

If at all, I think it would be better to have this handled by Scintilla when it already updates the current line style. This would probably much more efficient.

So, I suggest to open a feature request at Scintilla first, @advice2020.

**Comment by advice2020** [comment]
@nyamatongwe
Hey Neil, here is one of the posts I mentioned, if you have time to read this.
Thank You
